module/zfs/spa_history.c

```diff
@@ -408,7 +414,7 @@ spa_history_log_nvl(spa_t *spa, nvlist_t *nvl)
 
 	/* Kick this off asynchronously; errors are ignored. */
 	dsl_sync_task_nowait(spa_get_dsl(spa), spa_history_log_sync,
-	    nvarg, 0, ZFS_SPACE_CHECK_NONE, tx);
+	    spa_history_log_free, nvarg, 0, ZFS_SPACE_CHECK_NONE, tx);
```

Comment:
With the addition of `spa_history_log_free()` it should no longer be necessary to use `ZFS_SPACE_CHECK_NONE`.  This can be changed to `ZFS_SPACE_CHECK_RESERVED` or arguably `ZFS_SPACE_CHECK_EXTRA_RESERVED` to minimize the chance of history entries being dropped.